### PR TITLE
feat(goal_planner): align vehicle footprint heading parallel to parking side lane boundary

### PR DESCRIPTION
**Pull request by soblin**
## Description

fix the goal pose orientation same as the pose of closest pose on reference path to the ego footprint's front heading

## Related links

**Parent Issue:**

- https://tier4.atlassian.net/browse/RT1-7456

## How was this PR tested?

https://evaluation.tier4.jp/evaluation/reports/a55c06d7-9aec-5e8b-9f4d-b4b9dfc90799?project_id=prd_jt

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/CMakeLists.txt [hunk 1]
```diff
@@ -3,7 +3,7 @@ project(autoware_behavior_path_goal_planner_module)
 
 set(ament_cmake_lint_cmake_FOUND TRUE)
 
-option(BUILD_EXAMPLES "Build examples" OFF)
+option(BUILD_EXAMPLES "Build examples" ON)
```
Comment (soblin) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/CMakeLists.txt:6]: turn this OFF

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_searcher.cpp [hunk 2]
```diff
@@ -193,8 +196,23 @@ GoalCandidates GoalSearcher::search(const std::shared_ptr<const PlannerData> & p
...
+         transformed_vehicle_footprint.at(vehicle_info_utils::VehicleInfo::FrontRightIndex)) /
+        2.0;
+      lanelet::ConstLanelet vehicle_front_closest_lanelet;
+      lanelet::utils::query::getClosestLanelet(
+        pull_over_lanes, search_pose, &vehicle_front_closest_lanelet);
+      const auto vehicle_front_pose_for_bound = goal_planner_utils::calcClosestPose(
+        left_side_parking_ ? vehicle_front_closest_lanelet.leftBound()
+                           : vehicle_front_closest_lanelet.rightBound(),
+        autoware::universe_utils::createPoint(
+          vehicle_front_midpoint.x(), vehicle_front_midpoint.y(), search_pose.position.z));
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/goal_searcher.cpp:212]:
```suggestion
      const auto vehicle_front_pose_for_bound = goal_planner_utils::calcClosestPose(
        left_side_parking_ ? vehicle_front_closest_lanelet.leftBound()
                           : vehicle_front_closest_lanelet.rightBound(),
        autoware::universe_utils::createPoint(
          vehicle_front_midpoint.x(), vehicle_front_midpoint.y(), search_pose.position.z));
      if (!vehicle_front_pose_for_bound.has_value()) continue;
```

or set search_pose.orientation is better?

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/util.cpp [hunk 3]
```diff
@@ -740,4 +740,85 @@ lanelet::Lanelet createDepartureCheckLanelet(
...
+  const double sign = left_side_parking ? -1.0 : 1.0;
+  const double offset_from_center_line =
+    sign * (distance_from_bound.value() + parameters.margin_from_boundary);
+
+  const auto refined_goal_pose = calcOffsetPose(center_pose, 0, -offset_from_center_line, 0);
+
+  return refined_goal_pose;
+}
+
+Pose calcClosestPose(lanelet::ConstLineString3d line, const Point & query_point)
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/util.cpp:796]:
```suggestion
std::optional<Pose> calcClosestPose(const lanelet::ConstLineString3d line, const Point & query_point)
```

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/util.cpp [hunk 4]
```diff
@@ -740,4 +740,85 @@ lanelet::Lanelet createDepartureCheckLanelet(
...
+{
+  const auto segment =
+    lanelet::utils::getClosestSegment(lanelet::BasicPoint2d{query_point.x, query_point.y}, line);
+  if (segment.empty()) {
+    return geometry_msgs::msg::Pose{};
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/src/util.cpp:801]:
```suggestion
    return {};
```

File: planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/CMakeLists.txt [hunk 5]
```diff
@@ -27,7 +27,7 @@ if(BUILD_EXAMPLES)
   FetchContent_Declare(
     matplotlibcpp17
     GIT_REPOSITORY https://github.com/soblin/matplotlibcpp17.git
-    GIT_TAG main
+    GIT_TAG master
```
Comment (kosuke55) [inline planning/behavior_path_planner/autoware_behavior_path_goal_planner_module/CMakeLists.txt:30]: This is the reason why you can't clone in CI ?

**Pull request merged**